timers/timers.go

```diff
@@ -0,0 +1,164 @@
+package timers
+
+import (
+	"runtime"
+	"sync"
+	"time"
+	"unsafe"
+
+	"github.com/uber-go/atomic"
+)
+
+const (
+	cacheline = 64
+
+	// State machine for timers.
+	stateDefault = iota
+	stateExpired
+	stateCanceled
+)
+
+type mutex struct {
+	sync.Mutex
+	// Avoid false sharing:
+	// http://mechanical-sympathy.blogspot.com/2011/07/false-sharing.html
+	pad [cacheline - unsafe.Sizeof(sync.Mutex{})]byte
+}
+
+// Timer FIXME
+type Timer struct {
+	f        func()
+	deadline uint64 // in ticks
+	state    atomic.Int32
+	prev     *Timer
+}
+
+// Stop FIXME
+func (t *Timer) Stop() {
+	t.state.Store(stateCanceled)
+}
+
+func (t *Timer) append(tail *Timer) *Timer {
+	tail.prev = t
+	return tail
+}
+
+// Wheel FIXME
+type Wheel struct {
```

Comment:
Will try a struct with both a slice and its backing array next; switching to pooled slices from a linked list has actually made things ~30ns slower.